packages/now-cli/src/commands/build/index.ts

```diff
@@ -0,0 +1,183 @@
+  return (await globP(pattern, options)).map((p: string) =>
+    p.replace(/\\/g, '/')
+  );
+};
+
+const cwd = process.cwd();
+const outputDir = path.join(cwd, '.now');
+const workDir = path.join(outputDir, 'workDir');
+const buildersDir = path.join(outputDir, '/builders');
+const buildsOutputDir = path.join(outputDir, 'builds');
+
+async function getBuilds(
+  files: string[],
+  output: Output
+): Promise<Builder[] | null> {
+  let builds: Builder[] = [];
+  let nowJson;
+
+  try {
+    nowJson = await fs.readJson(path.join(cwd, 'now.json'));
+    const toResolve: Builder[] = [];
+    builds = (nowJson.builds || []).filter((build: Builder) => {
+      if (_glob.hasMagic(build.src)) {
+        toResolve.push(build);
+        return false;
+      }
+      return true;
+    });
+
+    for (const build of toResolve) {
+      const results = await glob(build.src, { cwd, nodir: true });
+      results.forEach((src: string) => {
+        builds.push({
+          src,
+          use: build.use,
+          config: build.config
+        });
+      });
+    }
+  } catch (err) {
+    output.debug('No now.json found, using zero-config');
+  }
+
+  if (builds.length === 0) {
+    let pkg;
+    try {
+      pkg = await fs.readJson(path.join(cwd, 'package.json'));
+    } catch (err) {
+      output.debug('No package.json found');
+    }
+    const { builders } = await detectBuilders(files, pkg);
+    builds.push(...(builders || []));
+  }
+
+  return builds;
+}
+
+export default async function main(ctx: NowContext) {
+  const debug = ctx.argv.some(arg => arg === '-d' || arg === '--debug');
+  const onlyArgIdx = ctx.argv.indexOf('--only');
+  const onlyBuild =
+    onlyArgIdx !== -1 && ctx.argv[ctx.argv.indexOf('--only') + 1];
+
+  const output = createOutput({ debug });
+  output.log('Setting up builds...');
+
+  const files = (await getAllProjectFiles(cwd, output))
+    .concat(
+      // getAllProjectFiles doesn't include dotfiles (.babelrc) so grab those
+      await glob('**/.*', { cwd, nodir: true })
+    )
+    .filter(file => {
+      return !file.startsWith('node_modules') && !file.startsWith('.now');
+    });
+
+  const fileRefs: { [filePath: string]: FileFsRef } = {};
+
+  for (const fsPath of files) {
+    const relPath = path.relative(cwd, fsPath);
+    const { mode } = await fs.stat(fsPath);
+    fileRefs[relPath] = new FileFsRef({ mode, fsPath });
+  }
+  let builds = await getBuilds(files, output);
+
+  if (builds && onlyBuild) {
+    output.debug(`Filtering by src: ${onlyBuild}`);
+    builds = builds.filter(build => build.src === onlyBuild);
+  }
+  if (!builds || builds.length === 0) {
+    return output.warn('No builds found');
+  }
+  output.log(`Found ${plural('build', builds.length, true)}`);
+  output.debug(`builds: ${JSON.stringify(builds)}`);
+
+  output.log('Installing builders');
+  const yarnDir = await getYarnPath(output);
+  const packagesSet = new Set<string>(builds.map(build => build.use));
```

Comment:
You need to special case `@now/static` which is not a real builder so it does not need to be installed.

Reply:
Added handling of `@now/static` 👍